cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/ApplicationMeta.java

```diff
@@ -28,10 +30,19 @@ public class ApplicationMeta {
 
   private final String id;
   private final ApplicationSpecification spec;
+  private final long creationTimeMillis;
```

Comment:
Does it make more sense to be named `updateTimeMillis` if we have to have this field here and we're using it in `updateAppSpec(ApplicationId appId, ApplicationSpecification spec)` ?

Reply:
so we view  this as the creationTime for the app version - it's main purpose is to identify the the version of the app (can be also called version metadata) and not to updateAppSpec - we just pass in the value to updateAppSpec - since this cannot be null (as it is an index column) - but doesn't serve any purpose in that function